extensions-core/multi-stage-query/src/main/java/org/apache/druid/msq/querykit/WindowOperatorQueryKit.java

```diff
@@ -201,41 +267,46 @@ private boolean ifEmptyOverPresentInWindowOperstors(
         operatorFactoryList = new ArrayList<>();
       } else if (of instanceof NaivePartitioningOperatorFactory) {
         if (((NaivePartitioningOperatorFactory) of).getPartitionColumns().isEmpty()) {
+          // TODO: This logic need to be revamped in the future. We probably don't need to handle empty over() cases separately.
```

Comment:
please don't leave `TODO`-s in the final version of the PR; 
it's fine to use them while you work on the PR to mark places you intent to get back (I use the same approach with `FIXME` tags)

If at some point its something you will not address in the actual patch; you could create an issue and remove the `TODO`.

Reply:
Done